src/posix/tty.rs

```diff
@@ -82,6 +83,35 @@ impl TTYPort {
         Ok(port)
     }
 
+    pub fn list_ports() -> Vec<SerialPortInfo> {
+        let context = libudev::Context::new().unwrap();
+        let mut enumerator = libudev::Enumerator::new(&context).unwrap();
+        enumerator.match_subsystem("tty").unwrap();
+        let mut vec = Vec::new();
+        for device in enumerator.scan_devices().unwrap() {
+            // Skip devices that don't have a parent. It's unclear why this works,
+            // but libserialport does it in their enumeration and it seems to work well.
+            let parent = match device.parent() {
+                Some(p) => p,
+                None => continue
+            };
+
+            // For serial8250 ports, try to open them (through their parent) assuming they're
+            // virtual if you can't.
+            let name = String::from(device.syspath().to_str().unwrap());
+            if parent.driver() == Some(OsStr::new("serial8250")) {
+                match ::open(&name) {
```

Comment:
As I mentioned in an earlier comment, I don't think it's safe to open random devices on a user's computer.